core/src/main/java/com/linecorp/armeria/internal/common/stream/SurroundingPublisher.java

```diff
@@ -349,7 +351,11 @@ public void onError(Throwable cause) {
         @Override
         public void onComplete() {
             setState(State.REQUIRE_BODY, State.REQUIRE_TAIL);
-            publish();
+            if (tail != null) {
+                publish();
+            } else {
+                close0(null);
+            }
```

Comment:
✅ updated, I found 1 more bug by unit-test.

If not `close() on tail == null`, below scenario failed

```
// elements: 1, head -> empty() -> end (no tail)
1. request(1)
2. sendHead() // requested: 1 -> 0
3. upstream.subscribe() -> onComplete() // empty body
4. publish() -> requested: 0 -> return; // expectComplete(), but no onComplete()
```
On (4), we can send `downstream.onComplete()` immediately cause `tail == null`!